src/lib/sample_metadata.rs

```diff
@@ -325,10 +326,20 @@ pub fn validate_samples(
         // then add in the undetermined sample.  The ordinal must be larger than the maximum
         // ordinal of an existing sample.
         let undetermined_ordinal = samples.iter().map(|s| s.ordinal).max().unwrap() + 1;
+        let lane: Option<usize> = {
+            let sample_lanes: Vec<Option<usize>> =
+                samples.iter().map(|s| s.lane).unique().collect();
+            if sample_lanes.len() == 1 {
+                sample_lanes[0]
+            } else {
+                None
+            }
+        };
```

Comment:
I thought it was better to find the lane information from the remaining samples.  I could use:

```
 let lane: Option<usize> = if lanes.len() == 1 { Some(lanes[0]) } else { None };
```

but that wouldn't work if `lanes` is empty and the sample sheet had specified the same lane across all remaining samples.